python/cudf_polars/cudf_polars/experimental/io.py

```diff
@@ -257,33 +265,116 @@ def do_evaluate(
         )
 
 
-def _sample_pq_statistics(ir: Scan) -> dict[str, np.floating[T]]:
-    import itertools
-
+def _sample_pq_statistics(ir: Scan) -> TableStats:
     import numpy as np
+    import pyarrow.compute as pa_c
+    import pyarrow.dataset as pa_ds
 
     # Use average total_uncompressed_size of three files
-    n_sample = min(3, len(ir.paths))
-    metadata = plc.io.parquet_metadata.read_parquet_metadata(
-        plc.io.SourceInfo(random.sample(ir.paths, n_sample))
-    )
-    column_sizes = {}
-    rowgroup_offsets_per_file = np.insert(
-        np.cumsum(metadata.num_rowgroups_per_file()), 0, 0
-    )
+    # TODO: Use plc.io.parquet_metadata.read_parquet_metadata
```

Comment:
Do you think what we need / is missing from pylibcudf is stable enough to write up an issue (if we haven't already)?